autogen/agentchat/conversable_agent.py

```diff
@@ -877,13 +888,17 @@ def _summarize_chat(
                 summary = self._reflection_with_llm(prompt, msg_list, llm_agent=agent, cache=cache)
             except BadRequestError as e:
                 warnings.warn(f"Cannot extract summary using reflection_with_llm: {e}", UserWarning)
-        elif method == "last_msg" or method is None:
+        elif summary_method == "last_msg" or summary_method is None:
             try:
                 summary = agent.last_message(self)["content"].replace("TERMINATE", "")
             except (IndexError, AttributeError) as e:
                 warnings.warn(f"Cannot extract summary using last_msg: {e}", UserWarning)
+        elif isinstance(summary_method, Callable):
+            summary = summary_method(recipient, self)
         else:
-            warnings.warn(f"Unsupported summary method: {method}", UserWarning)
+            warnings.warn(
```

Comment:
Shall we raise here? And add test coverage.

Reply:
You mean raise an error?

Reply:
yes

Reply:
You mean raise an error?